### DISCUSSION TITLE
Crud service limit constraint behaviour

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#376] [General]
### Description

Hi!
I wanted to ask you If I got the correct behavior of the env CRUD_LIMIT_CONSTRAINT_ENABLED.

This variable sets the default limit to 25 records. If you specify in the query another value uses the specified one, but, if you choose a value bigger than 200, you will be truncated to 200 records. This is because the default max limit is 200 and you can modify it with the env CRUD_MAX_LIMIT. Correct?

If I set the env CRUD_LIMIT_CONSTRAINT_ENABLED to false, I don't have any limit at all.

Now there is a question: is there a way to change the default limit? like if I wanted to keep the CRUD_LIMIT_CONSTRAINT_ENABLED set to true but I want to retrieve as default at most 50 items.

Thanks

**Comment by danibix95** [comment]
Hi, from my understanding of the [service code](https://github.com/mia-platform/crud-service/blob/main/lib/schemaGetters.js#L204), your reasoning is correct. `CRUD_LIMIT_CONSTRAINT_ENABLED` act as a flag to select whether the limit is active, whereas `CRUD_MAX_LIMIT` specify such limit.

Currently it seems there's no manner to customize the default limit value.

**Reply by elmalakomar** [reply]
Ok! Do you think it is feasible to add a new environment variable used to customize the default limit value? 
It could be a better solution that completely remove the limits for some projects.
